client/src/js/wall/FirstUser.js

```diff
@@ -1,67 +1,78 @@
-        return (
-            <WallContainer>
-                <WallDialog size="lg">
-                    <BoxGroupHeader>
-                        <h2>Setup User</h2>
-                        <p>Create an initial administrative user to start using Virtool.</p>
-                    </BoxGroupHeader>
-                    <form onSubmit={this.handleSubmit}>
+    return (
+        <WallContainer>
+            <WallDialog size="lg">
+                <BoxGroupHeader>
+                    <h2>Setup User</h2>
+                    <p>Create an initial administrative user to start using Virtool.</p>
+                </BoxGroupHeader>
+                <Formik initialValues={initialValues} onSubmit={handleSubmit}>
+                    <Form>
                         <BoxGroupSection>
                             <InputGroup>
                                 <InputLabel>Username</InputLabel>
-                                <Input name="username" value={username} onChange={this.handleChange} />
+                                <Field type="text" name="username" as={Input} />
+                                {usernameErrors.map(error => (
+                                    <InputError key={error}>{error}</InputError>
+                                ))}
                             </InputGroup>
                             <InputGroup>
                                 <InputLabel>Password</InputLabel>
-                                <PasswordInput name="password" value={password} onChange={this.handleChange} />
+                                <Field name="password" as={PasswordInput} />
+                                {passwordErrors.map(error => (
+                                    <InputError key={error}>{error}</InputError>
+                                ))}
                             </InputGroup>
                         </BoxGroupSection>
                         <WallDialogFooter>
                             <Button type="submit" icon="user-plus" color="blue">
                                 Create User
                             </Button>
+                            <InputError>{generalError}</InputError>
                         </WallDialogFooter>
-                    </form>
-                </WallDialog>
-            </WallContainer>
-        );
-    }
-}
+                    </Form>
+                </Formik>
+            </WallDialog>
+        </WallContainer>
+    );
+};
 
 export const mapDispatchToProps = dispatch => ({
     onSubmit: (username, password) => {
         dispatch(createFirstUser(username, password));
     }
 });
 
-export default connect(noop(), mapDispatchToProps)(FirstUser);
+export const mapStateToProps = state => ({
+    generalError: get(state, "errors.CREATE_FIRST_USER_ERROR.message", ""),
+    usernameErrors: get(state, "errors.CREATE_FIRST_USER_ERROR.errors.user_id", [""]),
+    passwordErrors: get(state, "errors.CREATE_FIRST_USER_ERROR.errors.password", [""])
+});
```

Comment:
How do you feel about this:
```js
export const mapStateToProps = state => ({
  errors: {
    general: get(state, "errors.CREATE_FIRST_USER_ERROR.message", ""),
    username: get(state, "errors.CREATE_FIRST_USER_ERROR.errors.user_id", [""]),
    password: get(state, "errors.CREATE_FIRST_USER_ERROR.errors.password", [""])
  }
});
```

Reply:
I think that is a better structure for the state, and will also help make the input props appear much lighter.